### PR TITLE
add sqs developer endpoints at /_aws/sqs

### PR DESCRIPTION
**Pull request by thrau**
This PR adds developer endpoints for SQS to `/_aws/sqs`. Specifically, it adds an endpoint that is compatible with `ReceiveMessage` that can be used to retrieve all messages of a queue without side-effects. To simplify things, it returns all message attributes and system attributes by default and ignores any other parameters from the `ReceiveMessage` operation, except `QueueUrl`.

This is a more robust replacement for features building on #7331 (/cc @lukqw)

Here are some example invocations:

```bash
# uses /_aws/sqs/messages as endpoint URL - demonstrating the compatibility with aws clients
aws --endpoint-url=http://localhost:4566/_aws/sqs/messages \
    sqs receive-message \
    --queue-url http://queue.localhost.localstack.cloud:4566/000000000000/test-queue-01

# equivalent cURL call but as GET request
curl "http://localhost:4566/_aws/sqs/messages?QueueUrl=http://queue.localhost.localstack.cloud:4566/000000000000/test-queue-01"

# there's also an endpoint where the queue information is encoded into the path directly
curl -H 'Accept: application/json' "http://localhost:4566/_aws/sqs/messages/us-east-1/000000000000/test-queue-01"
```

TODO:
- [X] basic implementation
- [x] write tests

### CODE REVIEW

**Comment by thrau** [comment]
cc @baermat

File: tests/integration/test_sqs_backdoor.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    """
+    d = xmltodict.parse(xml)
+
+    return [
+        {attr["Name"]: attr["Value"] for attr in msg["Attribute"]}
+        for msg in d["ReceiveMessageResponse"]["ReceiveMessageResult"]["Message"]
+    ]
+
+
+class TestSqsDeveloperEdpoints:
```
Comment (bentsku) [inline tests/integration/test_sqs_backdoor.py:23]: nit: typo
```suggestion
class TestSqsDeveloperEndpoints:
```

File: tests/integration/test_sqs_backdoor.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        )
+        attributes = _parse_message_attributes(response.text)
+        assert attributes[0]["ApproximateReceiveCount"] == "0"
+        assert attributes[1]["ApproximateReceiveCount"] == "0"
+
+        # do a real receive op that has a side effect
+        response = sqs_client.receive_message(
+            QueueUrl=queue_url, VisibilityTimeout=0, MaxNumberOfMessages=1, AttributeNames=["All"]
+        )
+        print(response)
```
Comment (bentsku) [inline tests/integration/test_sqs_backdoor.py:43]: nit: leftover statement

File: localstack/services/sqs/provider.py [hunk 3]
```diff
@@ -88,6 +92,13 @@
 LOG = logging.getLogger(__name__)
 
 
+class InvalidAddress(ServiceException):
+    code = "InvalidAddress"
+    message = "The address https://queue.amazonaws.com/ is not valid for this endpoint."
```
Comment (bentsku) [inline localstack/services/sqs/provider.py:97]: nit: probably not very important as we don't aim for parity here, but is the address hardcoded? I guess it's the closest exception to what we want to convey, right? In that case I guess it does not really matter.

**Review by bentsku — APPROVED** [review]
This is awesome! LGTM!
This is a great, it's amazing what we can do now with the router, and with the added compatibility with the SDK clients 🤯
Only minor minor nits, nothing blocking. This is really great, can't wait to see how we can expand this kind of things to help us in the platform/improve the developer experience.
Thanks!

File: localstack/services/sqs/provider.py (hunk 3 above)
Comment (thrau) [reply localstack/services/sqs/provider.py:97]: Exactly, just aiming for parity! It's literally what comes out of `aws sqs` when you do anything with an invalid queue URL, regardless of address :shrug: .

**Pull request merged**